server/embedded/src/org/labkey/embedded/EmbeddedExtractor.java

```diff
@@ -78,30 +79,52 @@ private File verifyJar()
     private boolean shouldExtract(File webAppLocation)
     {
         File existingVersionFile = new File(webAppLocation, "WEB-INF/classes/VERSION");
-        File existingDistributionFile = new File(webAppLocation, "WEB-INF/classes/distribution");
+        File existingDistributionFile = new File(webAppLocation, "WEB-INF/classes/distribution.properties");
 
         LabKeyDistributionInfo incomingDistribution = getDistributionInfo();
 
-        // Fresh installation or upgrading from non-embedded Tomcat
+        // Fresh installation
         if (!existingVersionFile.exists() || !existingDistributionFile.exists())
         {
             LOG.info("Extracting new LabKey distribution - %s".formatted(incomingDistribution));
             return true;
         }
```

Comment:
This is not necessarily a fresh distribution. This could also mean we're upgrading from a pre-`distribution.properties` embedded distribution.
This will correctly trigger extraction but it won't log any information about the existing distribution. Maybe it's not worth the trouble to log more info when upgrading like that.

Reply:
Updated comment